client/lxd_operations.go

```diff
@@ -77,6 +77,19 @@ func (r *ProtocolLXD) GetOperationWait(uuid string, timeout int) (*api.Operation
 	return &op, etag, nil
 }
 
+// GetOperationWaitSecret returns an Operation entry for the provided uuid and secret once it's complete or hits the timeout
+func (r *ProtocolLXD) GetOperationWaitSecret(uuid string, secret string, timeout int) (*api.Operation, string, error) {
+	op := api.Operation{}
+
+	// Fetch the raw value
+	etag, err := r.queryStruct("GET", fmt.Sprintf("/operations/%s/wait/secret/%s?timeout=%d", url.PathEscape(uuid), url.PathEscape(secret), timeout), nil, "", &op)
```

Comment:
Hmm, what is this URL?

I would have expected `/operations/%s/wait?secret=%s&timeout=%d` here